File: client/client/DTO/FilterRequestDto.cs
using client.DTO.Offer;

namespace client.DTO;

public class FilterRequestDto {
    public int? MinOfferPrice { get; set; }
    public int? MaxOfferPrice { get; set; }
    public DateTime? PickupTimeStart { get; set; }
    public DateTime? PickupTimeEnd { get; set; }
    public List<string>? Categories { get; set; }
    
    public LocationDto? Location { get; set; }
    
    public String? TextSearch { get; set; }
    
    
   
}

File: server/src/main/java/com/example/server/dto/offer/FilterRequestDto.java
package com.example.server.dto.offer;

import lombok.Data;

import java.time.LocalDateTime;
import java.util.List;

@Data
public class FilterRequestDto {
    private Integer minOfferPrice;
    private Integer maxOfferPrice;
    private LocalDateTime pickupTimeStart;
    private LocalDateTime pickupTimeEnd;
    private List<String> categories;
    private LocationDto location;
}
